hugolib/multilingual.go

```diff
@@ -0,0 +1,48 @@
+package hugolib
+
+import (
+	"github.com/spf13/cast"
+	"github.com/spf13/viper"
+)
+
+type Multilingual struct {
+	enabled bool
+	config  *viper.Viper
+
+	Languages []string
+}
+
+func (ml *Multilingual) GetString(key string) string { return cast.ToString(ml.Get(key)) }
+func (ml *Multilingual) GetStringMap(key string) map[string]interface{} {
+	return cast.ToStringMap(ml.Get(key))
+}
+
+func (ml *Multilingual) GetStringMapString(key string) map[string]string {
+	return cast.ToStringMapString(ml.Get(key))
+}
+
+func (ml *Multilingual) Get(key string) interface{} {
```

Comment:
Last time I thought of those things, I tried to limit the things that would be multi-lingual aware, knowing we could easily add some more, but it would be difficult to remove some once people depend on them.

I remember some parameters would break across language switching, and that production of certain links to other languages required the same config for all language to make sense. I'm not sure about `paginatePath`.. but we could add it if it's missing, useful and doesn't break cross-language links generation.